kite-data/kite-data-core/src/main/java/org/kitesdk/data/Datasets.java

```diff
@@ -606,6 +606,75 @@ public static boolean delete(String uri) {
   }
 
   /**
+   * Move a {@link Dataset} identified by the given dataset URI to trash.
+   * <p>
+   * When you call this method using a dataset URI, both data and metadata are
+   * moved to trash. After you call this method, the dataset no longer exists, unless
+   * an exception is thrown.
+   * <p>
+   * When you call this method using a view URI, data in that view is moved to trash.
+   * The dataset's metadata is not changed. This can throw an
+   * {@code UnsupportedOperationException} if the move to trash requires additional
+   * work. For example, if some, but not all, of the data in an underlying data
+   * file must be removed, then the implementation is allowed to reject the
+   * move to trash rather than copy the remaining records to a new file.
+   * An implementation must document under what conditions it accepts moves to trash,
+   * and under what conditions it rejects them.
+   * <p>
+   * URIs must begin with {@code dataset:}. The remainder of
+   * the URI is implementation specific, depending on the dataset scheme.
+   *
+   * @param uri a {@code Dataset} URI
+   * @return {@code true} if any data or metadata is removed, {@code false}
+   *          otherwise
+   * @throws NullPointerException if {@code uri} is null
+   * @throws IllegalArgumentException if {@code uri} is not a dataset URI
+   */
+  public static boolean moveToTrash(URI uri) {
```

Comment:
The logic in this method is a largely a copy of the delete(URI uri) method. Is it worth adding a private delete(URI uri, boolean useTrash) method that both call? I wouldn't really mind leaving it as is, since a small amount of redundant code might be simpler than factoring out a method.